libs/wire-api-federation/src/Wire/API/Federation/BackendNotifications.hs

```diff
@@ -31,7 +32,8 @@ data BackendNotification = BackendNotification
     -- this body, which could be very large and completely useless to the
     -- pusher. This also makes development less clunky as we don't have to
     -- create a sum type here for all types of notifications that could exist.
-    body :: RawJson
+    body :: RawJson,
+    requestId :: Maybe RequestId
```

Comment:
Why is this an optional field?

Reply:
Where is this actually set? I get that in theory when parsing a header it might be missing, but my understanding is we should make it mandatory, i.e., part of every request. If that is the case, then it'd be nice if this can also be reflected in types.

Reply:
It's set by nginx and it's written into the `Env` as part of the servant middleware.

Reply:
OK, I thought so. I'd prefer if we abort in the middleware if there's no request ID, but maybe that's too much.

I'll leave this up to you.

Reply:
I hear you. The only purpose of the request ID is debugging. If we made it a required header, we would have to change all of our integration tests. IMO this is not justified.